addon/index.d.ts

```diff
@@ -0,0 +1,759 @@
+   * with that value.
+   */
+  readonly value: T | null;
+
+  /**
+   * If this TaskInstance is canceled or throws an error (or yields
+   * a promise that rejects), this property will be set with that error.
+   * Otherwise, it is null.
+   */
+  readonly error: unknown;
+
+  /**
+   * True if the task instance is fulfilled.
+   */
+  readonly isSuccessful: boolean;
+
+  /**
+   * True if the task instance resolves to a rejection.
+   */
+  readonly isError: boolean;
+
+  /**
+   * True if the task instance was canceled before it could run to completion.
+   */
+  readonly isCanceled: boolean;
+
+  /**
+   * True if the task instance has started, else false.
+   */
+  readonly hasStarted: boolean;
+
+  /**
+   * True if the task has run to completion.
+   */
+  readonly isFinished: boolean;
+
+  /**
+   * True if the task is still running.
+   */
+  readonly isRunning: boolean;
+
+  /**
+   * Describes the state that the task instance is in. Can be used for debugging,
+   * or potentially driving some UI state. Possible values are:
+   *
+   * - `"dropped"`: task instance was canceled before it started
+   * - `"canceled"`: task instance was canceled before it could finish
+   * - `"finished"`: task instance ran to completion (even if an exception was thrown)
+   * - `"running"`: task instance is currently running (returns true even if
+   *     is paused on a yielded promise)
+   * - `"waiting"`: task instance hasn't begun running yet (usually
+   *     because the task is using the {@linkcode TaskProperty#enqueue .enqueue()}
+   *     task modifier)
+   *
+   * The animated timeline examples on the [Task Concurrency](/#/docs/task-concurrency)
+   * docs page make use of this property.
+   */
+  readonly state: 'dropped' | 'canceled' | 'finished' | 'running' | 'waiting';
+
+  /**
+   * True if the TaskInstance was canceled before it could
+   * ever start running. For example, calling
+   * {@linkcode Task#perform .perform()} twice on a
+   * task with the {@linkcode TaskProperty#drop .drop()} modifier applied
+   * will result in the second task instance being dropped.
+   */
+  readonly isDropped: boolean;
+
+  /**
+   * Cancels the task instance. Has no effect if the task instance has
+   * already been canceled or has already finished running.
+   *
+   * @param cancelReason Defaults to `".cancel() was explicitly called"`.
+   */
+  cancel(cancelReason?: string): void;
+
+  /**
+   * Returns a promise that resolves with the value returned
+   * from the task's (generator) function, or rejects with
+   * either the exception thrown from the task function, or
+   * an error with a `.name` property with value `"TaskCancelation"`.
+   */
+  then<TResult1 = T, TResult2 = never>(
+    onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | null,
+    onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | null
+  ): Promise<TResult1 | TResult2>;
+
+  catch<TResult = never>(
```

Comment:
This doesn't have docs?

Reply:
Should I add docs for this or remove it?

Reply:
Ahh, good catch. Yes this should be documented API.

Reply:
Actually, they are already "documented", in that the exists and are marked as public, just didn't have a text description.